File: server/docker/Dockerfile.cpp
# Sandbox runner image for C++ submissions.
# Build:  docker build -t nexthire-judge-cpp -f docker/Dockerfile.cpp .
# Then point JUDGE_IMAGE_CPP=nexthire-judge-cpp (optional; the default is gcc:13).
FROM gcc:13

RUN useradd -u 1000 -m runner
USER runner
WORKDIR /judge

File: server/docker/Dockerfile.java
# Sandbox runner image for Java submissions.
# Build:  docker build -t nexthire-judge-java -f docker/Dockerfile.java .
# Then point JUDGE_IMAGE_JAVA=nexthire-judge-java (optional; the default is eclipse-temurin:17-jdk).
FROM eclipse-temurin:17-jdk

RUN useradd -u 1000 -m runner
USER runner
WORKDIR /judge
